core/src/main/kotlin/in/specmatic/core/utilities/ContractSource.kt

```diff
@@ -117,12 +117,12 @@ data class GitRepo(
             if (!sourceDir.exists())
                 sourceDir.mkdirs()
 
-            if (!sourceGit.workingDirectoryIsGitRepo()) {
+            if (!sourceGit.workingDirectoryIsGitRepo() || (sourceGit.workingDirectoryIsGitRepo() && sourceGit.getRemoteUrl() != this.gitRepositoryURL && sourceDir.listFiles()?.isEmpty()!!)) {
```

Comment:
Can we extract `(sourceGit.workingDirectoryIsGitRepo() && sourceGit.getRemoteUrl() != this.gitRepositoryURL && sourceDir.listFiles()?.isEmpty()!!)` to a method that is named such that it conveys the purpose of these conditional checks? Just to aid readability.

Reply:
Thanks for the feedback, i'm fixing this right now